accepted/0000-salt-new-maintaining-workflow.md

```diff
@@ -0,0 +1,252 @@
+
+All these steps needs to be performed manually, with the help of some tooling, to eventually create a manual Submit Request to our Salt package in OBS.
+
+With Salt Extensions appearing now in the upcoming Salt 3008 release, we want to introduce a new workflow that suits better and solves some the limitations we currently have. 
+
+The purpose of this RFC is:
+- Define an new workflow for Salt that reduces user intervention to zero after a given PR is merged in `openSUSE/salt` repository until getting the package ready to consume at OBS.
+- Make the Salt maintaining workflow aligned with openSUSE practices.
+- Provide a workflow that can also work the same way with packaged Salt Extensions.
+- Deprecate the usage of `salt-packaging` repository.
+ 
+# Detailed design
+[design]: #detailed-design
+
+In this new workflow the `openSUSE/salt` GitHub repository will become the unique source of thrust, and it will contain:
+
+- Salt codebase
+- Packaging artifacts: spec file, changelog and extra sources
+- OBS workflow file
+
+Taking inspiration from the [OBS/SCM integration guide](https://openbuildservice.org/help/manuals/obs-user-guide/cha-obs-scm-ci-workflow-integration), the new workflow will use OBS workflows and GitHub Webhooks to automate pulling the changes from GitHub to OBS.
+
+In addition to this, as new Jenkins job will take care of making the OBS package ready to be submitted to openSUSE or SLE.
+
+This is how the proposed OBS structure would look like:
+
+- `systemsmanagement:saltstack/salt`:
+   * no services enabled - package ready to be submitted to openSUSE or SLE.
+   * branched from `systemsmanagement:saltstack:github/salt`
+- `systemsmanagement:saltstack:github/salt`:
+   * services enabled
+   * package building based on `openSUSE/release/xxxx` GitHub branch.
+- `systemsmanagement:saltstack:github:CI:...:PR-XXXX/salt`:
+   * services enabled
+   * package building according to PR branch.
+   * branched and removed automatically from `systemsmanagement:saltstack:github/salt` by OBS workflow.
+
+The same OBS structure will apply to all our OBS targets: `products`, `products:testing` and `products:next`.
+
+### Packaging artifacts
+
+All current extra "Sources" files in our RPM package, together with spec file and changelog file will go now to a `pkg/suse/` directory in `openSUSE/salt`:
+
+```
+pkg/suse/README.SUSE
+pkg/suse/html.tar.bz2
+pkg/suse/salt-tmpfiles.d
+pkg/suse/transactional_update.conf
+pkg/suse/update-documentation.sh
+pkg/suse/salt.spec
+pkg/suse/salt.changes
+```
+
+This is the place now where all those files will be maintained.
+
+#### Salt RPM changelog
+
+As mentioned this is now at `pkg/suse/salt.changes` in `openSUSE/salt` GitHub repo.
+
+When creating a PR to `openSUSE/salt` the user must also include the corresponding changes to the spec file, that can be generated as usual with `osc vc`.
```

Comment:
Not sure if I get this point right. Maybe it says `spec file` but the actual meaning is different, not sure, could you please clarify it?

Reply:
Woops, sorry. I meant "changelog" file :smile:
Let me fix this

Reply:
Do I understand it right, that in this case we will add changelog entries manually to the changelog file or manually but with `osc vc`, still not fully clear here. With the osc services there is way to use commit messages as a changlog items, don't we want to use it this way?

Reply:
I'm proposing here to manually add the changelog entry to `salt.changes` by using the `osc vc` command directly your Git tree, so the generated changelog entry (and header) can be included together with the code changes in the PR to `openSUSE/salt` repo.

This way a single PR to `openSUSE/salt` repo could contain all possible different bits: code changes, specfile changes, changelog entries, artifacts changes.

See this example PR: https://github.com/meaksh/salt/pull/10/files

Now that you mentioned the "osc services" for the changelog, I realized that the current proposal is not covering the fact that we need to maintain different changelogs depending on the targeted codestreams (as they are not aligned)

I'll have some thoughts and clarify this on the RFC text.

Reply:
I've added more text to the RFC on how to deal with different maintained changelogs